language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/ImportsAcceptor.java

```diff
@@ -39,15 +40,16 @@
  * @since 1.3.0
  */
 public class ImportsAcceptor {
+
     private final Set<String> newImports;
     private final Map<ImportDeclarationNode, ModuleSymbol> currentModuleImportsMap;
-    private final BiConsumer<String, String> onExistCallback;
+    private final BiConsumer<String, CodeActionModuleId> onExistCallback;
```

Comment:
Isn't the Import Acceptor being used in other completion scenarios as well? Hence I think having operation specific data model violates the SRP. Instead of `CodeActionModuleId,` this should be something common.

Reply:
Though the name has `CodeAction` in it. The content of that class is not operation specific. I think it's just a naming convention issue we have. WDYT of renaming that class from `CodeActionModuleId` to something else?

Reply:
@IMS94 Exactly, we need to rename it to something common, as well as get it out of the codeaction module.

Reply:
+1

Reply:
Done